src/cljs/org/broadinstitute/firecloud_ui/utils.cljs

```diff
@@ -152,8 +153,10 @@
            :headers (merge {"Authorization" (str "Bearer " @access-token)}
                            (:headers arg-map))
            :on-done (fn [{:keys [status-code] :as m}]
-                      (when (not= @server-down? (contains? #{500 503} status-code))
-                        (swap! server-down? not))
+                      (when (and (not @server-down?)  (not @maintenance-mode?))
+                        (cond
+                          (= status-code 502) (swap! maintenance-mode? not)
```

Comment:
@dmohs I left the `swap maintenance-mode? not` here because Matt said he had issues with the modal popping up multiple times (apparently the server-watcher wasn't working as expected, it was getting triggered even if you were just keeping the value at true). But I got rid of it in the other places, and it seems to be working fine.